web/profiles/custom/yalesites_profile/modules/custom/ys_views_basic/src/Plugin/Field/FieldType/ViewsBasicParams.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Drupal\ys_views_basic\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\Field\FieldStorageDefinitionInterface as StorageDefinition;
+
+/**
+ * Plugin implementation of the 'views_basic_params' field type.
+ *
+ * @FieldType(
+ *   id = "views_basic_params",
+ *   label = @Translation("Views Basic Params"),
+ *   description = @Translation("Stores parameters to pass to Views"),
+ *   category = @Translation("Custom"),
+ *   default_widget = "views_basic_default_widget",
+ *   default_formatter = "views_basic_default_formatter"
+ * )
+ */
+class ViewsBasicParams extends FieldItemBase {
+
+  /**
+   * Field type properties definition.
+   */
+  public static function propertyDefinitions(StorageDefinition $storage) {
+
+    $properties = [];
+
+    $properties['params'] = DataDefinition::create('string')
+      ->setLabel(t('Parameters'));
+
+    return $properties;
+  }
+
+  /**
+   * Field type schema definition.
+   */
+  public static function schema(StorageDefinition $storage) {
+
+    $columns = [];
+    $columns['params'] = [
+      'default' => '',
+      'description' => 'Parameter data to be sent to Views',
+      'not null' => FALSE,
+      'serialize' => TRUE,
+      'size' => 'big',
+      'type' => 'text',
+      'type' => 'blob',
+    ];
+
+    return [
+      'columns' => $columns,
+      'indexes' => [],
```

Comment:
I think this is correct since we are storing a lot of complex data that does not make sense in isolation. It still always feels funny to me to create a table without an index; but I'm not sure anything would apply. Just thinking out loud.